trie-db/src/fatdb.rs

```diff
@@ -71,6 +73,14 @@ where
 	> {
 		FatDBIterator::<L>::new(&self.raw).map(|iter| Box::new(iter) as Box<_>)
 	}
+
+	fn key_iter<'a>(&'a self) -> Result<
+		Box<dyn TrieIterator<L, Item = TrieKeyItem<TrieHash<L>, CError<L>>> + 'a>,
+		TrieHash<L>,
+		CError<L>,
+	> {
+		unimplemented!()
```

Comment:
why is it ok not to implement it ?

Reply:
Both fatdb and sectriedb (same for the mut variants) are specific to ethereum client.
It certainly could be implemented just by added the key iterator variant struct, but it is not very helpfull with ethereum (do not use a hashed value). So the only reason is to avoid a few line of code, I can add it or we can wait until someone needs it.

Reply:
yes ok, in general I think that having half implemented trait is very error prone, even more when it is not documented, maybe we could have a new trait for this key iteration stuff, and only implement it where we need it.

Reply:
looking again, I think we should just implement it, even in a non efficient way is fine. by using iter underneath.

Reply:
My main issue was to add unneeded code to the codebase, but I did implement it for sectrie, so https://github.com/paritytech/trie/pull/142/commits/99e83d3272879790c44bce0b2619b1ab4cc642f3 implement that.